playground/miden-wasm/src/lib.rs

```diff
@@ -63,16 +63,61 @@ pub fn program(asm: &str, inputs_frontend: &str, output_count: u16) -> Vec<u64>
     )
     .map_err(|err| format!("Failed to generate exection trace = {:?}, and advice_tape = {:?}", err, advice_tape)).unwrap();
     
-    // TODO: Investigate why proof verification fails when outputCount > 1
-    //assert!(miden::verify(*program.hash(), &[], &outputs, proof).is_ok());
-    // tracked in https://github.com/0xPolygonMiden/examples/issues/19
     trace.program_outputs().stack_outputs(output_count as usize).to_vec()
 }
 
+/// Proves the program with the given inputs
+#[wasm_bindgen]
+pub fn prove_program(asm: &str, inputs_frontend: &str, output_count: u16) -> Vec<u64> {
+    console_error_panic_hook::set_once();
+    wasm_logger::init(wasm_logger::Config::default());
+
+    let assembler = Assembler::new();
+    let program = assembler.compile(&asm).expect("Could not compile source");
+    
+    let mut stack_init = <Vec<u64>>::new();
+    let mut advice_tape = <Vec<u64>>::new();
+
+    // TODO: We must catch all cases of inputs. But maybe in the frontend already.
+    if inputs_frontend.trim().is_empty() == false {
+        let inputs_des: InputFile = serde_json::from_str(&inputs_frontend)
+        .map_err(|err| format!("Failed to deserialize input data - {}", err)).unwrap();
+        
+        stack_init = get_stack_init(&inputs_des);
+        advice_tape = get_advice_tape(&inputs_des);
+    } 
+
+    let inputs = get_program_inputs(&stack_init, &advice_tape); 
+
+    // default (96 bits of security)
+    let proof_options = ProofOptions::with_96_bit_security();
+
+    let (output, _proof) = miden::prove(
+        &program,
+        &inputs,
+        &proof_options,
+    )
+    .map_err(|err| format!("Failed to generate exection trace = {:?}, and advice_tape = {:?}", err, advice_tape)).unwrap();
+    
+    // We convert the outputs into one Vector because we can't return tuples in wasm-bindgen.
+    // So the first n elements are the stack_outputs and the last n elements are the overflow_addrs.
+    // TODO: We can somehow refactor to return structures in wasm-bindgen. But I didn't find a way yet.
+    // There might be a way to return the ProgramOutputs struct even.
+    let mut stack_outputs = output.stack_outputs(output_count as usize).to_vec();
+    let mut overflow_addrs = output.overflow_addrs().to_vec();
+    stack_outputs.append(&mut overflow_addrs);
+    
+    // TODO: We must return trace_info as well, but we can not handle that yet in Miden v0.3.
+    // TODO: We must return the proof as well. But we can not handle that yet in the frontend.
+
+    stack_outputs
+
+}
+
 /// Basic test for the program
 #[wasm_bindgen_test]
-fn run_program() {
-    let output = program(
+fn test_run_program() {
+    let output = run_program(
```

Comment:
Does it make sense to add a basic test for proving too?

Reply:
Hmm indeed, we should add that.

Reply:
added a test